hoomd/md/EvaluatorPairSMie.h

```diff
@@ -0,0 +1,211 @@
+                v["m"] = sm4;
+
+                Scalar sigma = fast::pow(sm1 / sm2, 1 / (sm3 - sm4));
+                Scalar epsilon = sm1 / fast::pow(sigma, sm3) * (sm3 - sm4) / sm3 * fast::pow(sm3 / sm4, sm4 / (sm4 - sm3));
+
+                v["epsilon"] = epsilon;
+                v["sigma"] = sigma;
+
+                v["Delta"] = sm5;
+                return v;
+                }
+            #endif
+            }
+            #ifdef SINGLE_PRECISION
+            __attribute__((aligned(8)));
+            #else
+            __attribute__((aligned(16)));
+            #endif
+
+        //! Constructs the pair potential evaluator
+        /*! \param _rsq Squared distance between the particles
+            \param _rcutsq Squared distance at which the potential goes to 0
+            \param _n First, larger exponent that captures hard-core repulsion
+            \param _m Second, smaller exponent that captures attraction
+            \param _params Per type pair parameters of this potential
+            \param _Delta Horizontal shift in r
+        */
+        DEVICE EvaluatorPairSMie(Scalar _rsq, Scalar _rcutsq, const param_type& _params)
+            : rsq(_rsq), rcutsq(_rcutsq), sm1(_params.sm1), sm2(_params.sm2), sm3(_params.sm3), sm4(_params.sm4), sm5(_params.sm5)
+            {
+            }
+
+        //! SMie doesn't use diameter
+        DEVICE static bool needsDiameter() { return false; }
+        //! Accept the optional diameter values
+        /*! \param di Diameter of particle i
+            \param dj Diameter of particle j
+        */
+        DEVICE void setDiameter(Scalar di, Scalar dj) { }
+
+        //! SMie doesn't use charge
+        DEVICE static bool needsCharge() { return false; }
+        //! Accept the optional diameter values
+        /*! \param qi Charge of particle i
+            \param qj Charge of particle j
+        */
+        DEVICE void setCharge(Scalar qi, Scalar qj) { }
+
+        //! Evaluate the force and energy
+        /*! \param force_divr Output parameter to write the computed force divided by r.
+            \param pair_eng Output parameter to write the computed pair energy
+            \param energy_shift If true, the potential must be shifted so that V(r) is continuous at the cutoff
+            \note There is no need to check if rsq < rcutsq in this method. Cutoff tests are performed
+                  in PotentialPair.
+
+            \return True if they are evaluated or false if they are not because we are beyond the cutoff
+        */
+        DEVICE bool evalForceAndEnergy(Scalar& force_divr, Scalar& pair_eng, bool energy_shift)
+            {
+            // precompute some quantities
+            Scalar rcutinv = fast::rsqrt(rcutsq);
+            Scalar rcut = Scalar(1.0) / rcutinv;
+            Scalar r = fast::sqrt(rsq);
+            Scalar rinv = fast::rsqrt(rsq);
+
+            // compute the force divided by r in force_divr
+            if (r < (rcut + sm5) && sm1 != 0)
```

Comment:
```suggestion
            if (rsq < (rcutsq) && sm1 != 0)
```
The neighbor list is unaware of `\delta` so the r_cut condition cannot be modified here. If the user wishes to modify `r_cut` based on `\delta`, that is up to them when they set the parameter.

I think after making this change `rcutinv` and `rcut` are no longer needed.

Reply:
Ok yes, I don't think either `rcut` or `rcutinv` are required. I removed them, with tests passing.